### PR TITLE
chore: restrict search to prisma-client-js for previewFeatures telemetry

### PR DESCRIPTION
**Pull request by Jolg42**
Does that makes sense to you too?

Maybe it doesn't, for example there is a new community python client https://github.com/RobertCraigie/prisma-client-py

### CODE REVIEW

**Review by millsp — APPROVED** [review]

File: packages/cli/src/bin.ts [hunk 1]
```diff
@@ -198,15 +198,22 @@ async function main(): Promise<number> {
...
         schemaProvider = config.datasources[0].provider
       }
-      const generator = config.generators.find(
-        (gen) => gen.previewFeatures.length > 0,
-      )
+
+      // restrict the search to prisma-client-js and prisma-client-go previewFeatures
+      const generator = config.generators.find((generator) => {
+        return (
+          ['prisma-client-js', 'prisma-client-go'].includes(generator.name) &&
```
Comment (janpio) [inline packages/cli/src/bin.ts:209]:
```suggestion
          ['prisma-client-js'].includes(generator.name) &&
```

File: packages/generator-helper/src/types.ts [hunk 2]
```diff
@@ -75,13 +76,16 @@ export type BinaryPaths = {
...
   generator: GeneratorConfig
+  // TODO: what is otherGenerators for?
   otherGenerators: GeneratorConfig[]
   schemaPath: string
   dmmf: DMMF.Document
   datasources: DataSource[]
+  // TODO deprecate datamodel & rename to schema?
   datamodel: string
-  binaryPaths?: BinaryPaths
+  // TODO is it really always version hash? Feature is unclear.
```
Comment (janpio) [inline packages/generator-helper/src/types.ts:86]: probably engine version, so git commit hash?

Comment (Jolg42) [reply packages/generator-helper/src/types.ts:86]: It seems we use it for something and third party users use it for something else

File: packages/cli/src/bin.ts (hunk 1 above)
Comment (Jolg42) [reply packages/cli/src/bin.ts:209]: The unofficial Python client is named `prisma-client-py` and would be filtered as well as go with this PR/suggestion 

Maybe we just don't restrict the search here?

Comment (janpio) [reply packages/cli/src/bin.ts:209]: I think we only want the preview feature of the official JS Client here. Everything else is out of our control and does not give us a clear picture.

Comment (Jolg42) [reply packages/cli/src/bin.ts:209]: Then we want to keep `prisma-client-go` I would say

Comment (janpio) [reply packages/cli/src/bin.ts:209]: See link above: https://github.com/prisma/prisma/pull/8617#issuecomment-907216484 Go Client does not support any preview features, so if someone is providing them they have no effect.

Comment (Jolg42) [reply packages/cli/src/bin.ts:209]: Good point 👍🏼

File: packages/cli/src/bin.ts [hunk 3]
```diff
@@ -198,15 +198,22 @@ async function main(): Promise<number> {
...
-        (gen) => gen.previewFeatures.length > 0,
-      )
+
+      // restrict the search to prisma-client-js previewFeatures
+      const generator = config.generators.find((generator) => {
+        return (
+          generator.name === 'prisma-client-js' &&
+          generator.previewFeatures.length > 0
+        )
+      })
```
Comment (Jolg42) [inline packages/cli/src/bin.ts:212]:
```suggestion
      const generator = config.generators.find((generator) => generator.name === 'prisma-client-js' && generator.previewFeatures.length > 0)
```

File: packages/cli/src/bin.ts [hunk 4]
```diff
@@ -198,15 +198,18 @@ async function main(): Promise<number> {
...
       if (config.datasources.length > 0) {
         schemaProvider = config.datasources[0].provider
       }
-      const generator = config.generators.find(
-        (gen) => gen.previewFeatures.length > 0,
-      )
+
+      // restrict the search to prisma-client-js previewFeatures
+	  // this was not scoped to `prisma-client-js` before Prisma 3.0
+      const generator = config.generators.find((generator) => generator.name === 'prisma-client-js' && generator.previewFeatures.length > 0)
```
Comment (janpio) [inline packages/cli/src/bin.ts:208]: Do we have tests for this?

Comment (Jolg42) [reply packages/cli/src/bin.ts:208]: Nope 🎐

Comment (janpio) [reply packages/cli/src/bin.ts:208]: Scary. We should probably manually test CLI telemetry with a `dev` version then?

Comment (janpio) [reply packages/cli/src/bin.ts:208]: https://prisma-company.slack.com/archives/CC59J6VEY/p1630512455033900

**Pull request merged**